libmcount/dynamic.c

```diff
@@ -302,6 +303,11 @@ static int find_dynamic_module(struct dl_phdr_info *info, size_t sz, void *data)
 	struct uftrace_mmap *map;
 	bool is_executable = mcount_is_main_executable(info->dlpi_name, sym_info->filename);
 
+	if (fmd->skip_first) { /* skip main binary */
+		fmd->skip_first = false;
+		return !fmd->needs_modules;
```

Comment:
We have `mcount_is_main_executable()` and you should check the result value.

Reply:
Thanks, fixed